### ISSUE TITLE
False negative ``import-error`` when the package imported is one of pylint's dependency

### ISSUE THREAD

**Issue by Pierre-Sassoulas** [pylint-dev/pylint#10339]
### Bug description

```python
When linting with pylint one must install pylint in its environment to be able to lint. So when pylint is providing a dependency that is not set in the project packaging, pylint does not raise import-error appropriately. For example for yaml:


import yaml


Then in an environment where pyyaml is not available this will fail.
```

### Configuration

```ini

```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
No error.
```

### Expected behavior

E401: Unable to import yaml

### Pylint version

```shell
A problem since forever, I guess.
```

### Additional dependencies

```python

```

**Comment by pavan-msys** [comment]
Hi @Pierre-Sassoulas 

**Version:**
```
pylint 4.0.0-dev0
astroid 4.0.0a0
Python 3.10.12 (main, Feb  4 2025, 14:57:36) [GCC 11.4.0]
```
**OS:**
`Ubuntu 22.04`

**I tested this in my environment, and it's working as expected**

![Image](https://github.com/user-attachments/assets/90b2810f-6049-46ec-8223-3ad5d37a8926)
